libpages/response_writer_wrappers.go

```diff
@@ -0,0 +1,44 @@
+package libpages
+
+import "net/http"
+
+var _ http.ResponseWriter = responseWriterWrapper{}
+
+type writeOverride interface {
+	Bytes(req *http.Request) []byte
+}
+
+// responseWriterWrapper wraps a http.ResponseWriter, allowing overriding
+// response for selected status codes. This is mainly for custom error pages.
+//
+// This type is not goroutine safe.
```

Comment:
I don't see this used anywhere yet. Where/how are you planning on using it?

Reply:
This was a half done file and accidentally committed; sorry! Will remove from this branch.